common/logger/src/diem_logger.rs

```diff
@@ -118,6 +118,34 @@ impl LogEntry {
             message,
         }
     }
+
+    pub fn metadata(&self) -> Metadata {
+        self.metadata.clone()
+    }
+
+    pub fn thread_name(&self) -> Option<String> {
+        self.thread_name.clone()
+    }
+
+    pub fn backtrace(&self) -> Option<String> {
+        self.backtrace.clone()
+    }
+
+    pub fn hostname(&self) -> Option<&'static str> {
+        self.hostname.clone()
+    }
+
+    pub fn timestamp(&self) -> String {
+        self.timestamp.clone()
+    }
+
+    pub fn data(&self) -> BTreeMap<&'static str, serde_json::Value> {
+        self.data.clone()
+    }
+
+    pub fn message(&self) -> Option<String> {
+        self.message.clone()
+    }
```

Comment:
Let's avoid cloning.

Use `.as_deref()` for `Option<String>` types to convert it to `Option<&str>` and `.as_ref()` to convert `Option<&'static str>` to `Option<&str>` and just return a reference to the BTreeMap and Metadata instead of cloning.